app/models/validated_pb_core.rb

```diff
@@ -32,4 +33,25 @@ def method_validate
     return if errors.empty?
     fail 'Method validation errors: ' + errors.join("\n")
   end
+  
+  def url_validate
+    errors = []
+    PBCore.instance_methods(false).grep(/(src|url)s?/).each do |method|
+      urls = [send(method)].select {|u| u}.flatten
+      urls.each do |url|
+        begin
+          Curl::Easy.new(url).tap do |curl|
+            curl.http_head
+            unless curl.response_code == 200
+              errors << "HEAD #{url} (from ##{method}) not 200: #{curl.status}"
+            end
+          end
+        rescue => e
+          errors << (["'##{method}' failed: #{e.message}"] + e.backtrace[0..2]).join("\n")
+        end 
+      end
+    end
+    return if errors.empty?
+    fail 'Method validation errors: ' + errors.join("\n")
+  end
```

Comment:
Can we make this less magic, by explicitly stating which methods are being checked? E.g.

```ruby
  def url_validate
    errors = []
    validate_these_url_methods = [:outside_url, :aapb_url, :boston_tv_news_url,
                              :thumbnail_src, :proxy_srcs, :transcript_srcs]
    validate_these_url_methods.each do |method|
      urls = [send(method)].select {|u| u}.flatten
```